MLIR.cmake

```diff
@@ -1,30 +1,32 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Path to LLVM source folder.
-if(DEFINED ENV{LLVM_PROJ_SRC})
+if(DEFINED LLVM_PROJ_SRC)
```

Comment:
Is there a reason to have to separate statements both here and below? I think it would be better to say:
```if(NOT DEFINED LLVM_PROJ_SRC AND DEFINED ENV{LLVM_PROJ_SRC})```

Reply:
I'm not sure that I follow... isn't this equivalent to `elseif(DEFINED ENV{LLVM_PROJ_SRC})`?

Reply:
Right now you have:

```
if(DEFINED LLVM_PROJ_SRC)
elseif(DEFINED ENV{LLVM_PROJ_SRC})
```

I am suggesting that you change your statements to avoid the first (empty) if condition:

```
if(NOT DEFINED LLVM_PROJ_SRC AND DEFINED ENV{LLVM_PROJ_SRC})
  set(LLVM_PROJ_SRC $ENV{LLVM_PROJ_SRC})
elseif(NOT DEFINED LLVM_PROJ_SRC)
  message(FATAL_ERROR "env variable LLVM_PROJ_SRC not set")
endif()
```

or something similar.

Reply:
Do you mean this?
```
if(NOT DEFINED LLVM_PROJ_SRC AND DEFINED ENV{LLVM_PROJ_SRC})
  set(LLVM_PROJ_SRC $ENV{LLVM_PROJ_SRC})
endif()
if(NOT DEFINED LLVM_PROJ_SRC)
  message(FATAL_ERROR "env variable LLVM_PROJ_SRC not set")
endif()
```

Reply:
Looking closer at this, is seems like the condition has to appear somewhere in order to have the error message work, so it's not clear to me that changing this actually makes it easier to understand.